### PR TITLE
feat: implement client compact

### PR DESCRIPTION
**Pull request by bsbds**
based on: #311

Please briefly answer these questions:

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

    The compact method in `xline-client` can be implement after #311

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

**Comment by Phoenix500526** [comment]
Is this pr based on #399 ？

File: xline-client/src/types/kv.rs [hunk 1]
```diff
@@ -674,3 +674,51 @@ impl From<Txn> for xlineapi::TxnRequest {
         txn.req
     }
 }
+
+/// Compaction Request compacts the key-value store up to a given revision.
+/// All superseded keys with a revision less than the compaction revision will be removed.
```
Comment (Phoenix500526) [inline xline-client/src/types/kv.rs:679]: Compaction is not a simple deletion. It won't remove the latest revisions of keys even if their revisions are less than the compaction revision.

Comment (bsbds) [reply xline-client/src/types/kv.rs:679]: an extra note is added

Comment (Phoenix500526) [reply xline-client/src/types/kv.rs:679]: It's "Note that", not "Not that". You can use this as a comment: 
```suggestion
/// All keys with revisions less than the given revision will be compacted. The compaction process will remove all historical versions of these keys, except for the most recent one.  

/// For example, here is a revision list: [(A, 1), (A, 2), (A, 3), (A, 4), (A,5)]. We compact at revision 3. After the compaction, the revision list will become [ (A, 3), (A, 4), (A,5)]. All revisions less than 3 are deleted.  The latest revision, 3, will be kept. 
```

Comment (bsbds) [reply xline-client/src/types/kv.rs]: Sorry for the typo. the comment is updated

**Review by Phoenix500526 — APPROVED** [review]

**Review by iGxnon — APPROVED** [review]

**Pull request merged**